controller/src/main/java/io/pravega/controller/server/ControllerService.java

```diff
@@ -384,9 +387,24 @@ public CompletableFuture<CreateStreamStatus> createStream(String scope, String s
             return CompletableFuture.completedFuture(
                     CreateStreamStatus.newBuilder().setStatus(CreateStreamStatus.Status.INVALID_STREAM_NAME).build());
         }
+        try {
+            validateStreamConfig(streamConfig);
+        } catch (IllegalArgumentException e) {
+            log.error(requestId, "Create stream: {} failed due to invalid retention policy {}", stream, streamConfig);
+            return CompletableFuture.completedFuture(
+                    CreateStreamStatus.newBuilder().setStatus(CreateStreamStatus.Status.INVALID_RETENTION_POLICY).build());
+        }
         return callCreateStream(scope, stream, streamConfig, createTimestamp, requestId);
     }
 
+    @VisibleForTesting
+    protected void validateStreamConfig(StreamConfiguration streamConfig) {
+        boolean isInvalidStreamConfig =  Config.REQUIRE_RETENTION_POLICY
+                && !streamConfig.getScalingPolicy().equals(ScalingPolicy.fixed(1))
+                && streamConfig.getRetentionPolicy() == null;
+        Preconditions.checkArgument(!isInvalidStreamConfig, "Invalid retention policy for the stream");
```

Comment:
Same as above , we can make message more explicit.

Reply:
updated but it doesn't matter here as the calling method is handling error and we have a different log message there